beacon-chain/rpc/prysm/v1alpha1/validator/proposer_bellatrix.go

```diff
@@ -48,19 +48,41 @@ func (vs *Server) setExecutionData(ctx context.Context, blk interfaces.SignedBea
 	if err != nil {
 		log.WithError(err).Warn("Proposer: failed to check if builder can be used")
 	} else if canUseBuilder {
-		h, err := vs.getPayloadHeaderFromBuilder(ctx, slot, idx)
+		builderPayload, err := vs.getPayloadHeaderFromBuilder(ctx, slot, idx)
 		if err != nil {
 			builderGetPayloadMissCount.Inc()
 			log.WithError(err).Warn("Proposer: failed to get payload header from builder")
 		} else {
-			blk.SetBlinded(true)
-			if err := blk.SetExecution(h); err != nil {
-				log.WithError(err).Warn("Proposer: failed to set execution payload")
+			// Can only check values if block is Capella or later.
+			if blk.Version() >= version.Capella {
+				localPayload, err := vs.getExecutionPayload(ctx, slot, idx, blk.Block().ParentRoot(), headState)
+				if err != nil {
+					return errors.Wrap(err, "failed to get execution payload")
+				}
+				localValue, err := localPayload.Value()
+				if err != nil {
+					return errors.Wrap(err, "failed to get local payload value")
```

Comment:
Do we want to error and return in these examples? Why can't we fallback to a value of 0 and continue?

Reply:
Because this is a local block payload. If it fails then I argue we should abort the ship.
The ones that are ok to log and continue are builder block payload failures